order_guidelines.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: default
+---
+While in EBML there is no particular order for elements, in Matroska it is necessary to make sure some elements are placed in a certain order for better playback, seeking and editing efficiency. So far there was no guideline on the order of elements, and some existing programs may not feed these, but by following these guidelines your Matroska files should be optimum for playback in various conditions.
+
+# Level 1 elements
+
+Matroska only needs a few level 1 elements to be playable: Segment Info, Track Info, Clusters. These elements have to be present in a Matroska file. And the Segment Info and Track Info <em>must</em> always be found before the Clusters (either because they're located before the first Cluster or they can be found via Meta Seek elements; see below). All the other elements can be omitted although Cues (index) improve the playback experience greatly.
+
+After a Matroska file has been created it may still be edited. For example chapters, tags or cover art attachments can be added. To do that the Meta Seek needs to be updated and also some elements may be voided or extended. Therefore it's a good idea to use some padding after theses Elements. It is better to put that padding at the Level 1 so that they can easily be reshuffled without having to move them at the end.
+
+## Meta Seek
+
+When 1 Meta Seek Head is present, it should be the first element in a Segment as it should reference the other level 1 elements in that segment (except maybe the long list of Clusters). There is also the possibility that in some cases there are two Meta Seek Head sections like for files produced with mkvpropedit. In that case the first one is also placed first with only the position of the level 1 elements except the Clusters. The second Meta Seek is placed at the end and contains a lengthy list of all Clusters (and not the other level 1 elements). Placing the first Meta Seek Head other than the first position of the Segment would make it a lot less useful So it <em>must</em> be the first element of a Cluster.
```

Comment:
2016-05-18 7:34 GMT+02:00 Dave Rice <notifications@github.com>:

> In order_guidelines.md
> <https://github.com/Matroska-Org/matroska-specification/pull/4#discussion_r63647333>
> :
>
> > @@ -0,0 +1,47 @@
> > +---
> > +layout: default
> > +---
> > +While in EBML there is no particular order for elements, in Matroska it is necessary to make sure some elements are placed in a certain order for better playback, seeking and editing efficiency. So far there was no guideline on the order of elements, and some existing programs may not feed these, but by following these guidelines your Matroska files should be optimum for playback in various conditions.
> > +
> > +# Level 1 elements
> > +
> > +Matroska only needs a few level 1 elements to be playable: Segment Info, Track Info, Clusters. These elements have to be present in a Matroska file. And the Segment Info and Track Info <em>must</em> always be found before the Clusters (either because they're located before the first Cluster or they can be found via Meta Seek elements; see below). All the other elements can be omitted although Cues (index) improve the playback experience greatly.
> > +
> > +After a Matroska file has been created it may still be edited. For example chapters, tags or cover art attachments can be added. To do that the Meta Seek needs to be updated and also some elements may be voided or extended. Therefore it's a good idea to use some padding after theses Elements. It is better to put that padding at the Level 1 so that they can easily be reshuffled without having to move them at the end.
> > +
> > +## Meta Seek
> > +
> > +When 1 Meta Seek Head is present, it should be the first element in a Segment as it should reference the other level 1 elements in that segment (except maybe the long list of Clusters). There is also the possibility that in some cases there are two Meta Seek Head sections like for files produced with mkvpropedit. In that case the first one is also placed first with only the position of the level 1 elements except the Clusters. The second Meta Seek is placed at the end and contains a lengthy list of all Clusters (and not the other level 1 elements). Placing the first Meta Seek Head other than the first position of the Segment would make it a lot less useful So it <em>must</em> be the first element of a Cluster.
>
> I propose the initial commit mirror the current status of the associated
> page at http://matroska.org/technical/specs/index.html so we're starting
> from a familiar place and change from there.
>

I agree. Merged.


> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Matroska-Org/matroska-specification/pull/4/files/b638e6a2e2f9a812cef38c3ab32398d83846ad07#r63647333>
>



-- 
Steve Lhomme
Matroska association Chairman